### PR TITLE
KAFKA-7190: KIP-443; Remove streams overrides on repartition topics

### PR DESCRIPTION
**Pull request by guozhangwang**
As described in KIP-443 (https://cwiki.apache.org/confluence/display/KAFKA/KIP-443%3A+Return+to+default+segment.ms+and+segment.index.bytes+in+Streams+repartition+topics). We want to remove the aggressive overrides of segment.ms and segment.index.bytes for repartition topics. The remaining segment.bytes should still be effective in bounding its footprint.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -1003,17 +1003,12 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
         // verify that producer batch config is no larger than segment size, then add topic configs required for creating topics
         final Map<String, Object> topicProps = originalsWithPrefix(TOPIC_PREFIX, false);
+        final Map<String, Object> producerProps = getClientPropsWithPrefix(PRODUCER_PREFIX, ProducerConfig.configNames());
 
-        if (topicProps.containsKey(topicPrefix(TopicConfig.SEGMENT_INDEX_BYTES_CONFIG))) {
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:1007]: This is a piggy-backed fix to not use SEGMENT_INDEX_BYTES_CONFIG but SEGMENT_BYTES_CONFIG

**Comment by guozhangwang** [comment]
Triggered https://jenkins.confluent.io/job/system-test-kafka-branch-builder/2476/ .

**Review by bbejeck — APPROVED** [review]
Thanks @guozhangwang LGTM

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 2]
```diff
@@ -643,7 +643,7 @@ public void shouldAddInternalTopicConfigForRepartitionTopics() {
         final InternalTopologyBuilder.TopicsInfo topicsInfo = builder.topicGroups().values().iterator().next();
         final InternalTopicConfig topicConfig = topicsInfo.repartitionSourceTopics.get("appId-foo");
         final Map<String, String> properties = topicConfig.getProperties(Collections.<String, String>emptyMap(), 10000);
-        assertEquals(5, properties.size());
+        assertEquals(3, properties.size());
         assertEquals(String.valueOf(Long.MAX_VALUE), properties.get(TopicConfig.RETENTION_MS_CONFIG));
```
Comment (soenkeliebau) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:647]: I think this needs to be changed to -1 to fit with what is being set in the properties. At the moment this test fails consistently.

Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:647]: @soenkeliebau That was my bad, I ran than unit test while this change was stashed. Thanks!

**Pull request merged**